akka-cluster/src/main/scala/akka/cluster/ClusterHeartbeat.scala

```diff
@@ -23,19 +31,20 @@ import akka.util.ccompat._
  */
 @InternalApi
 @ccompatUsedUntil213
-private[cluster] final class ClusterHeartbeatReceiver extends Actor with ActorLogging {
+private[cluster] final class ClusterHeartbeatReceiver(getCluster: () => Cluster) extends Actor with ActorLogging {
   import ClusterHeartbeatSender._
 
   // Important - don't use Cluster(context.system) in constructor because that would
   // cause deadlock. See startup sequence in ClusterDaemon.
-  lazy val cluster = Cluster(context.system)
-  lazy val selfHeartbeatRsp = HeartbeatRsp(cluster.selfUniqueAddress)
+  lazy val cluster = getCluster()
+
   lazy val verboseHeartbeat = cluster.settings.Debug.VerboseHeartbeatLogging
 
-  def receive = {
-    case Heartbeat(from) =>
-      if (verboseHeartbeat) cluster.ClusterLogger.logDebug("Heartbeat from [{}]", from)
-      sender() ! selfHeartbeatRsp
+  def receive: Receive = {
+    case hb: Heartbeat =>
+      // TODO log the sequence nr once serializer is enabled
+      if (verboseHeartbeat) cluster.ClusterLogger.logDebug("Heartbeat from [{}]", hb.from)
```

Comment:
We can't yet log `hb.sequenceNr` here?

Reply:
It'll be 0 every time until we release the serializer